pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1864,10 +1871,11 @@ cdef class YearOffset(SingleConstructorOffset):
 
     def _get_offset_day(self, other) -> int:
         # override BaseOffset method to use self.month instead of other.month
-        # TODO: there may be a more performant way to do this
-        return get_day_of_month(
-            other.replace(month=self.month), self._day_opt
-        )
+        cdef:
```

Comment:
can you type other here? (dateteime), this can be nogil ?

Reply:
> can you type other here? 

this used to be typed as datetime, but that broke some tests bc apparently we pass `date` sometimes.  need to revisit.

> this can be nogil ?

i think in this context thats actually not helpful since this isnt being called in a loop, so it just adds overhead of un-acquiring and re-acquiring the gil

Reply:
ok interesting, can you note that somewhere

Reply:
followon ok